etcdserver/server.go

```diff
@@ -649,7 +651,7 @@ func (s *EtcdServer) run() {
 		sched.Stop()
 
 		// wait for gouroutines before closing raft so wal stays open
-		s.wg.Wait()
+		s.goWait()
```

Comment:
since it's rejecting goroutines now, would it be cleaner to do this?
```go
s.wgMu.Lock()
close(s.stopping)
s.wgMu.Unlock()
s.wg.Wait()

sched.Stop()
```

```go
goAttach() {
s.wgMu.RLock()
defer s.mgMu.RUnlock()
select {
case <-s.stopping:
    plog.Warning(...)
    return
default:
}
...
}
```